Library/Homebrew/utils/shared_audits.rb

```diff
@@ -92,6 +92,48 @@ def gitlab_release(user, repo, tag, formula: nil, cask: nil)
     "#{tag} is a GitLab pre-release."
   end
 
+  def codeberg_repo_data(user, repo)
+    @codeberg_repo_data ||= {}
+    @codeberg_repo_data["#{user}/#{repo}"] ||= begin
+      out, _, status = Utils::Curl.curl_output("https://codeberg.org/api/v1/repos/#{user}%2F#{repo}")
+      json = JSON.parse(out) if status.success?
+      json = nil if json&.dig("message")&.include?("404 Project Not Found")
+      json
+    end
+  end
+
+  def codeberg_release_data(user, repo, tag)
+    id = "#{user}/#{repo}/#{tag}"
+    @codeberg_release_data ||= {}
+    @codeberg_release_data[id] ||= begin
+      out, _, status = Utils::Curl.curl_output(
+        "https://codeberg.org/api/v1/repos/#{user}%2F#{repo}/releases/tags/#{tag}", "--fail"
```

Comment:
```suggestion
        "https://codeberg.org/api/v1/repos/#{user}/#{repo}/releases/tags/#{tag}", "--fail"
```

seems to break with %2F for me